### ISSUE TITLE
Pass function to ofType

### ISSUE THREAD

**Issue by nicholaswyoung** [redux-observable/redux-observable#71]
When a function (like the ones returned by [redux-actions](/acdlite/redux-actions) `createAction()`) are passed to `ofType()`, they should be converted via `toString()` automatically. String action specifiers would be passed onward, unmodified.

Current syntax:

``` javascript
const queueSomethingEpic = createAction('app/activities/epic')
const doSomethingEpic = action$ => action$.ofType(queueSomethingEpic.toString())
```

Proposed syntax:

``` javascript
const doSomethingEpic = action$ => action$.ofType(queueSomethingEpic)
```

This seems to be an emerging method in the react/redux community. I've found it a useful pattern. But does it belong in redux-observable?

**Comment by jayphelps** [comment]
@nicholaswyoung can you provide some additional examples of where this pattern is becoming popular? in redux-actions, the `actionFactory.toString()` behavior appears undocumented AFAICT

**Comment by jayphelps** [comment]
Also--if we supported this either with explicit `.toString()` or casting `'' + type` wouldn't we be breaking anyone who wants to use things other than strings, for types? Support for that was added to redux, like using Symbols.

Can you think of a way of maintaining both, without any side effects?

**Comment by jayphelps** [comment]
btw--you can inject this behavior pretty easily. You could even publish it as an npm package if you want to. A sort of redux-observable "plugin".

```js
// patch-redux-observable.js
import { ActionsObservable } from 'redux-observable';

const originalOfType = ActionsObservable.prototype.ofType;
ActionsObservable.prototype.ofType = function ofType(...types) {
  return originalOfType.apply(this, types.map(type => type.toString()));
};
```

Then import it in, as side effects in some common entry file, such as where you `createStore()`.

```js
import { createEpicMiddleware } from 'redux-observable';
import './patch-redux-observable';

// etc such as createStore()
```

**Comment by nicholaswyoung** [comment]
@jayphelps I'm fairly sure the OP in #10 is using a similar technique. It also looks like a [hot topic](https://github.com/reactjs/redux/issues/628) over on the redux issue tracker as well. I picked up the idea from @jeffbski, who also turned me on to this library. 

All that to say, I seem to be uniquely positioned among folks who favor this technique, perhaps to the point of existing inside my own echochamber.

The more I think about it, this probably does belong as a plugin. At least for the time being, while we're feeling out the benefits and pitfalls of this technique.

**Comment by jayphelps** [comment]
@nicholaswyoung that's currently my line of thinking as well. I hadn't seen this pattern prior--I'll give it a whirl in one of my smaller apps and see what I think.

Thanks for being practical about this. Keeping this open for now until we have a decision one way or the other.

**Comment by jeffbski** [comment]
I like using redux-actions to create action creators and their associated constants. One nice thing is that you have half the things to keep track of. You only need the redux-action and then you can get the type automagically from the function being coerced into a string. In many places it just works without thinking about it (auto coercing toString), so adding that into `ofType` would be wonderful. The less people have to think about stuff like that the better.

Also FYI, I have a PR into redux-actions which adds createActions that goes even further in tightening up the code and reducing boilerplate. It adds the ability to add a prefix namespace to your action types which is real useful. https://github.com/acdlite/redux-actions/pull/104

**Comment by jayphelps** [comment]
This was implemented in #348 and will be in the next release 🎉
